### ISSUE TITLE
broker stan data race

### ISSUE THREAD

**Issue by vtolstov** [micro/go-plugins#326]
```
==================
WARNING: DATA RACE
Write at 0x00c0086d6118 by goroutine 62:
  github.com/micro/go-plugins/broker/stan.(*stanBroker).Disconnect()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-plugins/broker/stan/stan.go:236 +0x13c
  github.com/micro/go-micro/server.(*rpcServer).Start.func2()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-micro/server/rpc_server.go:546 +0x4fe

Previous read at 0x00c0086d6118 by goroutine 59:
  github.com/micro/go-plugins/broker/stan.(*stanBroker).connect()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-plugins/broker/stan/stan.go:161 +0x221
  github.com/micro/go-plugins/broker/stan.(*stanBroker).reconnectCB()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-plugins/broker/stan/stan.go:122 +0x6b
  github.com/micro/go-plugins/broker/stan.(*stanBroker).reconnectCB-fm()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-plugins/broker/stan/stan.go:120 +0x69

Goroutine 62 (running) created at:
  github.com/micro/go-micro/server.(*rpcServer).Start()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-micro/server/rpc_server.go:504 +0x848
  github.com/micro/go-micro.(*service).Start()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-micro/service.go:80 +0x104
  github.com/micro/go-micro.(*service).Run()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/micro/go-micro/service.go:116 +0x50
  unistack.org/centralv2/server.runService()
      /home/vtolstov/devel/projects/centralv2/server/server.go:172 +0x5c

Goroutine 59 (finished) created at:
  github.com/nats-io/go-nats-streaming.(*conn).closeDueToPing()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/nats-io/go-nats-streaming/stan.go:541 +0x180
  github.com/nats-io/go-nats-streaming.(*conn).pingServer()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/nats-io/go-nats-streaming/stan.go:478 +0x3c1
  github.com/nats-io/go-nats-streaming.(*conn).pingServer-fm()
      /home/vtolstov/devel/projects/centralv2/vendor/github.com/nats-io/go-nats-streaming/stan.go:468 +0x41
```

**Comment by vtolstov** [comment]
fix in #327

**Comment by asim** [comment]
Your PR was merged
